builder/osc/bsusurrogate/step_register_omi.go

```diff
@@ -146,16 +147,21 @@ func (s *StepRegisterOMI) combineDevices(snapshotIDs map[string]string) []oscgo.
 }
 
 func copyToDeviceMappingImage(device osc.BlockDeviceMappingVmCreation) oscgo.BlockDeviceMappingImage {
+	log.Printf("Copydevice mapping image ")
 	deviceImage := oscgo.BlockDeviceMappingImage{
 		DeviceName:        device.DeviceName,
 		VirtualDeviceName: device.VirtualDeviceName,
 		Bsu: &oscgo.BsuToCreate{
 			DeleteOnVmDeletion: device.Bsu.DeleteOnVmDeletion,
-			Iops:               device.Bsu.Iops,
-			SnapshotId:         device.Bsu.SnapshotId,
```

Comment:
I think the previous code was the right thing to do. Just copy what you have. The big problem here is that `device` contains invalid data.

Reply:
that's why, I check if they contain valid data, if yes set the data, if not skip

Reply:
With the accTest, I can see that it's safer to pass this data to the querry only if there are not empty

Reply:
done